irohad/main/irohad.cpp

```diff
@@ -159,6 +172,62 @@ static shared_model::crypto::Keypair getKeypairFromFile(
              });
 }
 
+boost::optional<std::string> getEnvVar(std::string const& key)
+{
+    char const* val = getenv(key.c_str()); 
+    return val == NULL ? boost::optional<std::string>() : std::string(val);
+}
+
+boost::optional<IrohadConfig::DbConfig> getPostgresCredsFromEnv() {
+  
+  auto pg_host = getEnvVar("IROHA_POSTGRES_HOST");
+  auto pg_port_str = getEnvVar("IROHA_POSTGRES_PORT");
+  auto pg_user = getEnvVar("IROHA_POSTGRES_USER");
+  auto pg_pass = getEnvVar("IROHA_POSTGRES_PASSWORD");
+  auto pg_w_dbname = getEnvVar("IROHA_POSTGRES_DATABASE");
+  auto pg_m_dbname = getEnvVar("IROHA_POSTGRES_MAINTENANCE_DATABASE");
+
+  std::uint16_t pg_port;
+  if (pg_port_str.has_value()) {
+    auto pg_port_int = std::stoi(pg_port_str.value());
+    if (pg_port_int < std::numeric_limits<std::uint16_t>::min() ||
+        pg_port_int > std::numeric_limits<std::uint16_t>::max()) {
+      std::ostringstream os;
+      os << "The IROHA_POSTGRES_PORT environment variable value (" << pg_port_int
+          << ") is out of range.";
+      throw std::invalid_argument(std::move(os).str());
+    } else {
+      pg_port = static_cast<std::uint16_t>(pg_port_int);
+    }
+  } 
+
+  if (pg_host.has_value() and 
```

Comment:
So this works only when ALL ENV vars are set?

What if I want to override only one parameter? Let's be able to override only one param.

Reply:
@kuvaldini Something like this? => f911da97298a3ca63d337c574d7b19db9805c24e